HopsanGUI/Utilities/GUIUtilities.cpp

```diff
@@ -395,6 +395,21 @@ void windowFunction(QVector<double> &data, WindowingFunctionEnumT function, doub
             }
             rCa = 0.5;
             rCb = 1.5;
+            break;
+        }
+        case FlatTopWindow: {
+            int N = data.size()-1;
+            rCa = 0;
+            for(int n=0; n<=N && N>0; ++n) {
+                double w = 1.0 - 1.933*cos(2*M_PI*n/N) + 1.286*cos(4*M_PI*n/N) - 0.388*cos(6*M_PI*n/N) + 0.0322*cos(8*M_PI*n/N);
+                data[n] *= w;
+                rCa += w;
+                rCb += w*w;
+            }
+            rCa /= N;
```

Comment:
Maybe need protection for division by zero in rCa rCb and rC?, the last one should it be b or c?

Reply:
I added a check for empty data sets at the top of the function.

I believe the computations are correct. The formulas are:
<a href="https://www.codecogs.com/eqnedit.php?latex=C_a&space;=&space;\textup{mean}(\omega)&space;\\&space;C_b&space;=&space;\frac{\textup{mean}(\omega^2)}{\textup{mean}(\omega)^2}" target="_blank"><img src="https://latex.codecogs.com/gif.latex?C_a&space;=&space;\textup{mean}(\omega)&space;\\&space;C_b&space;=&space;\frac{\textup{mean}(\omega^2)}{\textup{mean}(\omega)^2}" title="C_a = \textup{mean}(\omega) \\ C_b = \frac{\textup{mean}(\omega^2)}{\textup{mean}(\omega)^2}" /></a>

Reply:
nevermind, I did not look closely enough, I though there were three out parameters rCa, rCb and rCc (but the last does note exist)